MonoGame.Framework.Content.Pipeline/ExternalTool.cs

```diff
@@ -27,13 +31,39 @@ public static int Run(string command, string arguments)
             return result;
         }
 
+        public static void RestoreDotnetTool(string command, string toolName)
+        {
+            string path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory);
+            if (CurrentPlatform.OS == OS.Linux)
+                path= Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "linux");
+            if (CurrentPlatform.OS == OS.MacOSX)
+                path= Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "osx");
+            if (Directory.Exists (Path.Combine(path, toolName))) 
+                return;
+            Directory.CreateDirectory (path);
+            var exe = CurrentPlatform.OS == OS.Windows ? "dotnet.exe" : "dotnet";
+            var dotnetRoot = Environment.GetEnvironmentVariable ("DOTNET_ROOT");
+            if (!string.IsNullOrEmpty(dotnetRoot)) {
+                exe = Path.Combine(dotnetRoot, exe);
+            }
+            Run (exe, $"tool {command} {toolName} --tool-path .", out string stdOut, out string stdErr,  workingDirectory: path);
+        }
+
+        public static void RestoreDotnetTools()
+        {
+            RestoreDotnetTool("install", Crunch);
+            RestoreDotnetTool("install", BasisU);
+        }
+
         /// <summary>
         /// Run a dotnet tool. The tool should be installed in a .config/dotnet-tools.json file somewhere in the project lineage.
         /// </summary>
         public static int RunDotnetTool(string toolName, string args, out string stdOut, out string stdErr, string stdIn=null, string workingDirectory=null)
         {
-            var finalizedArgs = toolName + " " + args;
-            return ExternalTool.Run("dotnet", finalizedArgs, out stdOut, out stdErr, stdIn, workingDirectory);
+            RestoreDotnetTools ();
```

Comment:
Running the restore / install process every time is gonna be tedious, we should cache the result of dotnet tools being restored once it was successfully done so.

Reply:
We check here https://github.com/MonoGame/MonoGame/pull/8527/files#diff-3f7b501f8d90c49eb02c352c09349a0f48f4c4fe74e0f0d025746ac3615bdd51R46 to see if its installed already

Reply:
Oh I see, totally missed it.

Tho I don't know if we should check have to run executables to check if the tool is installed, as right now the more tools we add, the slower they will all launch as for each tool separately we would have to run `dotnet.exe` once.

Reply:
I'll add an extra file which we write so we only install all the tools once.